Update README (#1248)
meilisearch/meilisearch-js: README.md

**Meilisearch JavaScript** is the Meilisearch API client for JavaScript developers.

**Meilisearch** is an open-source search engine. [Learn more about Meilisearch.](https://github.com/meilisearch/meilisearch)

## Table of Contents <!-- omit in toc -->

- [📖 Documentation](#-documentation)
- [🔧 Installation](#-installation)
- [🎬 Getting started](#-getting-started)
- [🤖 Compatibility](#-compatibility-with-meilisearch)
- [💡 Learn more](#-learn-more)

> - [⚙️ Contributing](#️-development-workflow-and-contributing)

Reviewer: I think the anchor should be `#-contributing` now, as the title of the section below also changed

- [📜 API Resources](#-api-resources)

## 📖 Documentation

This readme contains all the documentation you need to start using Meilisearch's Javascript SDK.

> For general information on how to use Meilisearch, refer to our [main documentation website](https://docs.meilisearch.com/learn/) and the [Meilisearch API Reference](https://docs.meilisearch.com/reference/api/).

Reviewer 2: What do we need to precise `main website` ?

Author: I'm not 100% sure it is necessary, but in short I am trying to establish the difference between Meilisearch's "main" documentation website (i.e. docs.meilisearch.com) and the SDK documentation (i.e. this readme).

So "main documentation" works in opposition to "This readme contains all the JS SDK documentation".

What do you think?

Reviewer 2: Mmmh, I understand the idea. But once you say `main documentation website` and another time `Meilisearch API reference`  which might lead to maybe more confusion. 
But I do like the idea that we are specific on the fact that we are not talking about the meilisearch-js documentation. Not sure it is necessary though.

I'm not helping here 😂 Sorry

Author: You actually are helping quite a bit, don't worry. Those are valid observations.

I'll keep "main documentation" (or something similar), but review the readme so references to both docs.meilisearch and SDK docs are uniform to reduce ambiguity.

## 🔧 Installation

We recommend installing `meilisearch-js` in your project with your package manager of choice.

If you use `npm`:

```sh
npm install meilisearch
```

If you prefer `yarn`:

```sh
yarn add meilisearch
```

`meilisearch-js` officially supports `node` versions >= 12 and <= 16.

> Instead of using a package manager, you may also import the library directly into your HTML via a CDN.

Reviewer 2: ```html
<script src="https://cdn.jsdelivr.net/npm/meilisearch@0.25.1/dist/bundles/meilisearch.umd.min.js"></script>
```

Should we add this?

Author: I like that idea. Do you think adding the CDN link here makes the import section down below redundant?

This readme contains all the documentation you need to start using this Meilisearch SDK.

For general information on how to use Meilisearch—such as our API reference, tutorials, guides, and in-depth articles—refer to our [main documentation website](https://docs.meilisearch.com/).

## 🔧 Installation

We recommend installing `meilisearch-js` in your project with your package manager of choice.

If you use `npm`:

```sh
npm install meilisearch
```

If you prefer `yarn`:

```sh
yarn add meilisearch
```

`meilisearch-js` officially supports `node` versions >= 12 and <= 16.

Instead of using a package manager, you may also import the library directly into your [HTML via a CDN](#include-script-tag).

### Run Meilisearch <!-- omit in toc -->

To use one our SDKS, you must first have a running Meilisearch instance. Consult our documentation for [instructions on how to download and launch Meilisearch](https://docs.meilisearch.com/reference/features/installation.html#download-and-launch).

### Import <!-- omit in toc -->

After installing `meilisearch-js`, you must import it into your application. There are many ways of doing that depending on your development environment.

> #### Import Syntax <!-- omit in toc -->

Author: @bidoubiwa, what do you think about changing this heading from "Import syntax" to something like ES6 modules"?

Author: Based on our call earlier:

```suggestion
#### `import` syntax <!-- omit in toc -->
```

Reviewer 2: Good idea !

**Meilisearch JavaScript** is the Meilisearch API client for JavaScript developers.

**Meilisearch** is an open-source search engine. [Learn more about Meilisearch.](https://github.com/meilisearch/meilisearch)

## Table of Contents <!-- omit in toc -->

- [📖 Documentation](#-documentation)
- [🔧 Installation](#-installation)
- [🎬 Getting started](#-getting-started)
- [🤖 Compatibility with Meilisearch](#-compatibility-with-meilisearch)
- [💡 Learn more](#-learn-more)
- [⚙️ Contributing](#️-contributing)
- [📜 API resources](#-api-resources)

## 📖 Documentation

This readme contains all the documentation you need to start using this Meilisearch SDK.

For general information on how to use Meilisearch—such as our API reference, tutorials, guides, and in-depth articles—refer to our [main documentation website](https://docs.meilisearch.com/).

## 🔧 Installation

We recommend installing `meilisearch-js` in your project with your package manager of choice.

If you use `npm`:

```sh
npm install meilisearch
```

If you prefer `yarn`:

```sh
yarn add meilisearch
```

> `meilisearch-js` officially supports `node` versions >= 12 and <= 16.

Reviewer 2: ```suggestion
`meilisearch-js` officially supports `node` versions >= 14 and <= 18.
```

recently updated (yesterday)

Instead of using a package manager, you may also import the library directly into your [HTML via a CDN](#include-script-tag).

### Run Meilisearch <!-- omit in toc -->

> To use one our SDKS, you must first have a running Meilisearch instance. Consult our documentation for [instructions on how to download and launch Meilisearch](https://docs.meilisearch.com/reference/features/installation.html#download-and-launch).

Reviewer 2: ```suggestion
To use one of our SDKs, you must first have a running Meilisearch instance. Consult our documentation for [instructions on how to download and launch Meilisearch](https://docs.meilisearch.com/reference/features/installation.html#download-and-launch).
```

Author: 🙏

./meilisearch --master-key=masterKey
```

> You can also install Meilisearch using [Homebrew](https://brew.sh), [APT](https://ubuntu.com/server/docs/package-management), or [Docker](https://www.docker.com/). Consult the official documentation for more information on [downloading and running Meilisearch](https://docs.meilisearch.com/reference/features/installation.html#download-and-launch).

Reviewer 2: I think we don't need to list all the ways to download meilisearch. Just the link to the documentation is nice.

What do you think?

Author: Always up for cutting down text.

What do you think about removing the example above describing how to install Meilisearch locally while we're at it? I'm a bit afraid that showing an example of local installs with `curl` might make users think this is our recommendation as the best way of running Meilisearch

Reviewer 3: I agree let's cut the text and the `curl` example a link to the doc will be good enough

Reviewer 4: Just the link to the docs is enough!

> #### Require Syntax <!-- omit in toc -->

Author: Also here, @bidoubiwa: from "Require syntax" to "CommonJS modules"?

It's possible I'm just old, but whenever I read this "Require syntax" I think about [requireJS](https://requirejs.org/)/AMD modules instead of Node-style CommonJS modules.

Author: Based on our call:

```suggestion
#### `require` syntax <!-- omit in toc -->
```

#### React Native <!-- omit in toc -->

To use `meilisearch-js` with React Native, you must also install [react-native-url-polyfill](https://www.npmjs.com/package/react-native-url-polyfill).

#### Deno<!-- omit in toc -->

> Usage in a deno environment:

Reviewer 2: ```suggestion
Usage in a Deno environment:
```

})()
```

> Tasks such as document addition always return a unique identifier. You can use this identifier `uid` to check the status (`enqueued`, `processing`, `succeeded` or `failed`) of a [task](https://docs.meilisearch.com/reference/api/tasks.html#get-task).

Reviewer 2: ```suggestion
Tasks such as document addition always return a unique identifier. You can use this identifier `taskUid` to check the status (`enqueued`, `processing`, `succeeded` or `failed`) of a [task](https://docs.meilisearch.com/reference/api/tasks.html#get-task).
```

}
```

### Using search parameters <!-- omit in toc -->

> `meilisearch-js` supports all [search parameters](https://docs.meilisearch.com/reference/features/search_parameters.html) described in our official documentation.

Reviewer 2: Is `official` important?

  ])
```

You only need to perform this operation once per index.

> Note that Meilisearch will rebuild your index whenever you update `filterableAttributes`. Depending on the size of your dataset, this might take considerable time. You can track the process using the [tasks API](https://docs.meilisearch.com/reference/api/tasks.html#get-task)).

Reviewer 2: ```suggestion
Note that Meilisearch rebuilds your index whenever you update `filterableAttributes`. Depending on the size of your dataset, this might take considerable time. You can track the process using the [tasks API](https://docs.meilisearch.com/reference/api/tasks.html#get-task)).
```

Not sure about that

Author: Not sure about your edit or about the info?

IMHO, your edit is pretty good and I want to commit it.

Reviewer 2: I wasn't sure if it was correct english

This repository also contains [more examples](./examples).

## ⚙️ Contributing

We welcome all contributions, big and small! If you want to know more about this SDK's development workflow or want to contribute to the repo, please visit our [contributing guidelines](/CONTRIBUTING.md) for detailed instructions.

## 📜 API resources

### Search <!-- omit in toc -->

#### [Make a search request](https://docs.meilisearch.com/reference/api/search.html)

```ts
client.index<T>('xxx').search(query: string, options: SearchParams = {}, config?: Partial<Request>): Promise<SearchResponse<T>>
```

#### [Make a search request using the GET method (slower than the search method)](https://docs.meilisearch.com/reference/api/search.html#search-in-an-index-with-get-route)

```ts
client.index<T>('xxx').searchGet(query: string, options: SearchParams = {}, config?: Partial<Request>): Promise<SearchResponse<T>>
```

### Documents <!-- omit in toc -->

#### [Add or replace multiple documents](https://docs.meilisearch.com/reference/api/documents.html#add-or-replace-documents)

```ts

> client.index.addDocuments(documents: Document<T>[]): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').addDocuments(documents: Document<T>[]): Promise<EnqueuedTask>
```

```

#### [Add or update multiple documents](https://docs.meilisearch.com/reference/api/documents.html#add-or-update-documents)

```ts
client.index.updateDocuments(documents: Array<Document<Partial<T>>>): Promise<EnqueuedTask>
```

#### [Add or update multiple documents in batches](https://docs.meilisearch.com/reference/api/documents.html#add-or-update-documents)

```ts
client.index.updateDocumentsInBatches(documents: Array<Document<Partial<T>>>, batchSize = 1000): Promise<EnqueuedTask[]>
```

#### [Get Documents](https://docs.meilisearch.com/reference/api/documents.html#get-documents)

```ts
client.index.getDocuments(params: getDocumentsParams): Promise<Document<T>[]>
```

#### [Get one document](https://docs.meilisearch.com/reference/api/documents.html#get-one-document)

```ts
client.index.getDocument(documentId: string): Promise<Document<T>>
```

#### [Delete one document](https://docs.meilisearch.com/reference/api/documents.html#delete-one-document)

```ts
client.index.deleteDocument(documentId: string | number): Promise<EnqueuedTask>
```

#### [Delete multiple documents](https://docs.meilisearch.com/reference/api/documents.html#delete-documents)

```ts

> client.index.deleteDocuments(documentsIds: string[] | number[]): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').deleteDocuments(documentsIds: string[] | number[]): Promise<EnqueuedTask>
```

```

#### [Delete all documents](https://docs.meilisearch.com/reference/api/documents.html#delete-all-documents)

```ts
client.index.deleteAllDocuments(): Promise<Types.EnqueuedTask>
```

### Tasks <!-- omit in toc -->

#### [Get task info using the client](https://docs.meilisearch.com/reference/api/tasks.html#get-all-tasks)

##### Task list
```ts
client.getTasks(): Promise<Result<Task[]>>
```

##### One task
```ts
client.getTask(uid: number): Promise<Task>
```

#### [Get task info using the index](https://docs.meilisearch.com/reference/api/tasks.html#get-all-tasks-by-index)

##### Task list

```ts

> client.index.getTasks(): Promise<Result<Task[]>>

Reviewer 2: ```suggestion
client.index('myIndex').getTasks(): Promise<Result<Task[]>>
```

```

##### One task

```ts

> client.index.getTask(uid: number): Promise<Task>

Reviewer 2: ```suggestion
client.index('myIndex').getTask(uid: number): Promise<Task>
```

```

#### Wait for one task

##### Using the client

```ts
client.waitForTask(uid: number, { timeOutMs?: number, intervalMs?: number }): Promise<Task>
```

##### Using the index

```ts
client.index.waitForTask(uid: number, { timeOutMs?: number, intervalMs?: number }): Promise<Task>
```

#### Wait for multiple tasks

##### Using the client

```ts
client.waitForTasks(uids: number[], { timeOutMs?: number, intervalMs?: number }): Promise<Result<Task[]>>
```

##### Using the index

```ts

> client.index.waitForTasks(uids: number[], { timeOutMs?: number, intervalMs?: number }): Promise<Result<Task[]>>

Reviewer 2: ```suggestion
client.index('myIndex').waitForTasks(uids: number[], { timeOutMs?: number, intervalMs?: number }): Promise<Result<Task[]>>
```

#### [Get all indexes in Index instances](https://docs.meilisearch.com/reference/api/indexes.html#list-all-indexes)

```ts
client.getIndexes(): Promise<Index[]>
```

#### [Get raw indexes in JSON response from Meilisearch](https://docs.meilisearch.com/reference/api/indexes.html#list-all-indexes)

```ts
client.getRawIndexes(): Promise<IndexResponse[]>
```

#### [Create a new index](https://docs.meilisearch.com/reference/api/indexes.html#create-an-index)

```ts
client.createIndex<T>(uid: string, options?: IndexOptions): Promise<EnqueuedTask>
```

#### Create a local reference to an index

```ts
client.index<T>(uid: string): Index<T>
```

#### [Get an index instance completed with information fetched from Meilisearch](https://docs.meilisearch.com/reference/api/indexes.html#get-one-index)

```ts
client.getIndex<T>(uid: string): Promise<Index<T>>
```

#### [Get the raw index JSON response from Meilisearch](https://docs.meilisearch.com/reference/api/indexes.html#get-one-index)

```ts
client.getRawIndex(uid: string): Promise<IndexResponse>
```

#### [Get an object with information about the index](https://docs.meilisearch.com/reference/api/indexes.html#get-one-index)

```ts

> client.index.getRawInfo(): Promise<IndexResponse>

Reviewer 2: ```suggestion
client.index('myIndex').getRawInfo(): Promise<IndexResponse>
```

```

#### [Update Index](https://docs.meilisearch.com/reference/api/indexes.html#update-an-index)

##### Using the client

```ts
client.updateIndex(uid: string, options: IndexOptions): Promise<EnqueuedTask>
```

##### Using the index object

```ts

> client.index.update(data: IndexOptions): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').update(data: IndexOptions): Promise<EnqueuedTask>
```

```

#### [Delete index](https://docs.meilisearch.com/reference/api/indexes.html#delete-an-index)

##### Using the client
```ts
client.deleteIndex(uid): Promise<void>
```

##### Using the index object
```ts

> client.index.delete(): Promise<void>

Reviewer 2: ```suggestion
client.index('myIndex').delete(): Promise<void>
```

```

#### [Get specific index stats](https://docs.meilisearch.com/reference/api/stats.html#get-stat-of-an-index)

```ts

> client.index.getStats(): Promise<IndexStats>

Reviewer 2: ```suggestion
client.index('myIndex').getStats(): Promise<IndexStats>
```

```

##### Return Index instance with updated information

```ts

> client.index.fetchInfo(): Promise<Index>

Reviewer 2: ```suggestion
client.index('myIndex').fetchInfo(): Promise<Index>
```

```

##### Get Primary Key of an Index

```ts

> client.index.fetchPrimaryKey(): Promise<string | undefined>

Reviewer 2: ```suggestion
client.index('myIndex').fetchPrimaryKey(): Promise<string | undefined>
```

```

### Settings <!-- omit in toc -->

#### [Get settings](https://docs.meilisearch.com/reference/api/settings.html#get-settings)

```ts

> client.index.getSettings(): Promise<Settings>

Reviewer 2: ```suggestion
client.index('myIndex').getSettings(): Promise<Settings>
```

```

#### [Update settings](https://docs.meilisearch.com/reference/api/settings.html#update-settings)

```ts

> client.index.updateSettings(settings: Settings): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').updateSettings(settings: Settings): Promise<EnqueuedTask>
```

```

#### [Reset settings](https://docs.meilisearch.com/reference/api/settings.html#reset-settings)

```ts

> client.index.resetSettings(): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').resetSettings(): Promise<EnqueuedTask>
```

```

### Synonyms <!-- omit in toc -->

#### [Get synonyms](https://docs.meilisearch.com/reference/api/synonyms.html#get-synonyms)

```ts

> client.index.getSynonyms(): Promise<object>

Reviewer 2: ```suggestion
client.index('myIndex').getSynonyms(): Promise<Synonyms>
```

```

#### [Update synonyms](https://docs.meilisearch.com/reference/api/synonyms.html#update-synonyms)

```ts

> client.index.updateSynonyms(synonyms: Synonyms): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').updateSynonyms(synonyms: Synonyms): Promise<EnqueuedTask>
```

```

#### [Reset synonyms](https://docs.meilisearch.com/reference/api/synonyms.html#reset-synonyms)

```ts

> client.index.resetSynonyms(): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').resetSynonyms(): Promise<EnqueuedTask>
```

```

### Stop-words <!-- omit in toc -->

#### [Get stop-words](https://docs.meilisearch.com/reference/api/stop_words.html#get-stop-words)

```ts

> client.index.getStopWords(): Promise<string[]>

Reviewer 2: ```suggestion
client.index('myIndex').getStopWords(): Promise<string[]>
```

```

#### [Update stop-words](https://docs.meilisearch.com/reference/api/stop_words.html#update-stop-words)

```ts

> client.index.updateStopWords(stopWords: string[] | null ): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').updateStopWords(stopWords: string[] | null ): Promise<EnqueuedTask>
```

```

#### [Reset stop-words](https://docs.meilisearch.com/reference/api/stop_words.html#reset-stop-words)

```ts

> client.index.resetStopWords(): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').resetStopWords(): Promise<EnqueuedTask>
```

```

### Ranking rules <!-- omit in toc -->

#### [Get ranking rules](https://docs.meilisearch.com/reference/api/ranking_rules.html#get-ranking-rules)

```ts

> client.index.getRankingRules(): Promise<string[]>

Reviewer 2: ```suggestion
client.index('myIndex').getRankingRules(): Promise<string[]>
```

```

#### [Update ranking rules](https://docs.meilisearch.com/reference/api/ranking_rules.html#update-ranking-rules)

```ts

> client.index.updateRankingRules(rankingRules: string[] | null): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').updateRankingRules(rankingRules: string[] | null): Promise<EnqueuedTask>
```

```

#### [Reset ranking rules](https://docs.meilisearch.com/reference/api/ranking_rules.html#reset-ranking-rules)

```ts

> client.index.resetRankingRules(): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').resetRankingRules(): Promise<EnqueuedTask>
```

```

### Distinct Attribute <!-- omit in toc -->

#### [Get distinct attribute](https://docs.meilisearch.com/reference/api/distinct_attribute.html#get-distinct-attribute)

```ts

> client.index.getDistinctAttribute(): Promise<string | void>

Reviewer 2: ```suggestion
client.index('myIndex').getDistinctAttribute(): Promise<string | void>
```

```

#### [Update distinct attribute](https://docs.meilisearch.com/reference/api/distinct_attribute.html#update-distinct-attribute)

```ts

> client.index.updateDistinctAttribute(distinctAttribute: string | null): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').updateDistinctAttribute(distinctAttribute: string | null): Promise<EnqueuedTask>
```

```

#### [Reset distinct attribute](https://docs.meilisearch.com/reference/api/distinct_attribute.html#reset-distinct-attribute)

```ts

> client.index.resetDistinctAttribute(): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').resetDistinctAttribute(): Promise<EnqueuedTask>
```

```

### Searchable attributes <!-- omit in toc -->

#### [Get searchable attributes](https://docs.meilisearch.com/reference/api/searchable_attributes.html#get-searchable-attributes)

```ts

> client.index.getSearchableAttributes(): Promise<string[]>

Reviewer 2: ```suggestion
client.index('myIndex').getSearchableAttributes(): Promise<string[]>
```

```

#### [Update searchable attributes](https://docs.meilisearch.com/reference/api/searchable_attributes.html#update-searchable-attributes)

```ts

> client.index.updateSearchableAttributes(searchableAttributes: string[] | null): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').updateSearchableAttributes(searchableAttributes: string[] | null): Promise<EnqueuedTask>
```

```

#### [Reset searchable attributes](https://docs.meilisearch.com/reference/api/searchable_attributes.html#reset-searchable-attributes)

```ts

> client.index.resetSearchableAttributes(): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').resetSearchableAttributes(): Promise<EnqueuedTask>
```

```

### Displayed attributes <!-- omit in toc -->

#### [Get displayed attributes](https://docs.meilisearch.com/reference/api/displayed_attributes.html#get-displayed-attributes)

```ts

> client.index.getDisplayedAttributes(): Promise<string[]>

Reviewer 2: ```suggestion
client.index('myIndex').getDisplayedAttributes(): Promise<string[]>
```

```

#### [Update displayed attributes](https://docs.meilisearch.com/reference/api/displayed_attributes.html#update-displayed-attributes)

```ts

> client.index.updateDisplayedAttributes(displayedAttributes: string[] | null): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').updateDisplayedAttributes(displayedAttributes: string[] | null): Promise<EnqueuedTask>
```

```

#### [Reset displayed attributes](https://docs.meilisearch.com/reference/api/displayed_attributes.html#reset-displayed-attributes)

```ts

> client.index.resetDisplayedAttributes(): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').resetDisplayedAttributes(): Promise<EnqueuedTask>
```

```

### Filterable attributes <!-- omit in toc -->

#### [Get filterable attributes](https://docs.meilisearch.com/reference/api/filterable_attributes.html#get-filterable-attributes)

```ts

> client.index.getFilterableAttributes(): Promise<string[]>

Reviewer 2: ```suggestion
client.index('myIndex').getFilterableAttributes(): Promise<string[]>
```

```

#### [Update filterable attributes](https://docs.meilisearch.com/reference/api/filterable_attributes.html#update-filterable-attributes)

```ts

> client.index.updateFilterableAttributes(filterableAttributes: string[] | null): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').updateFilterableAttributes(filterableAttributes: string[] | null): Promise<EnqueuedTask>
```

```

#### [Reset filterable attributes](https://docs.meilisearch.com/reference/api/filterable_attributes.html#reset-filterable-attributes)

```ts

> client.index.resetFilterableAttributes(): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').resetFilterableAttributes(): Promise<EnqueuedTask>
```

```

### Sortable attributes <!-- omit in toc -->

#### [Get sortable attributes](https://docs.meilisearch.com/reference/api/sortable_attributes.html#get-sortable-attributes)

```ts

> client.index.getSortableAttributes(): Promise<string[]>

Reviewer 2: ```suggestion
client.index('myIndex').getSortableAttributes(): Promise<string[]>
```

```

#### [Update sortable attributes](https://docs.meilisearch.com/reference/api/sortable_attributes.html#update-sortable-attributes)

```ts

> client.index.updateSortableAttributes(sortableAttributes: string[] | null): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').updateSortableAttributes(sortableAttributes: string[] | null): Promise<EnqueuedTask>
```

```

#### [Reset sortable attributes](https://docs.meilisearch.com/reference/api/sortable_attributes.html#reset-sortable-attributes)

```ts

> client.index.resetSortableAttributes(): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').resetSortableAttributes(): Promise<EnqueuedTask>
```

```

### Typo tolerance <!-- omit in toc -->

#### [Get typo tolerance](https://docs.meilisearch.com/reference/api/typo_tolerance.html#get-typo-tolerance)

```ts

> client.index.getTypoTolerance(): Promise<TypoTolerance>

Reviewer 2: ```suggestion
client.index('myIndex').getTypoTolerance(): Promise<TypoTolerance>
```

```

#### [Update typo tolerance](https://docs.meilisearch.com/reference/api/typo_tolerance.html#update-typo-tolerance)

```ts

> client.index.updateTypoTolerance(typoTolerance: TypoTolerance | null): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').updateTypoTolerance(typoTolerance: TypoTolerance | null): Promise<EnqueuedTask>
```

```

#### [Reset typo tolerance](https://docs.meilisearch.com/reference/api/typo_tolerance.html#reset-typo-tolerance)

```ts

> client.index.resetTypoTolerance(): Promise<EnqueuedTask>

Reviewer 2: ```suggestion
client.index('myIndex').resetTypoTolerance(): Promise<EnqueuedTask>
```

<hr>

> Meilisearch provides and maintains many SDKs and Integration tools like this one. We want to provide everyone with an **amazing search experience for any kind of project**. For a full overview of everything we create and maintain, take a look at the [integration-guides](https://github.com/meilisearch/integration-guides) repository.

Reviewer 4: Should we use `Integration` or just `integration`?

Reviewer 2: should we make it plural?

Author: You mean in "Meilisearch provides and maintains many SDKs and Integration tools […]", correct?

I'm 100% with writing "integration tools" in lowercase. We shouldn't make integrations plural, though—in this case, "integration" works as an adjective for "tools", so plural would be wrong (like "fats cats").

Reviewer 2: I agree !
